src/Aspire.Hosting/Keycloak/KeycloakBuilderExtension.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Publishing;
+
+namespace Aspire.Hosting;
+
+public static class KeycloakBuilderExtensions
+{
+    private const int DefaultContainerPort = 8080;
+
+    public static IResourceBuilder<KeycloakContainerResource> AddKeycloakContainer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null)
+    {
+        var keycloakContainer = new KeycloakContainerResource(name);
+
+        return builder
+            .AddResource(keycloakContainer)
+            .WithManifestPublishingCallback(WriteKeycloakContainerToManifest)
+            .WithAnnotation(new EndpointAnnotation(ProtocolType.Tcp, uriScheme: "http", port: port, containerPort: DefaultContainerPort))
+            .WithAnnotation(new ContainerImageAnnotation { Image = "quay.io/keycloak/keycloak", Tag = "latest" })
+            .WithEnvironment("KEYCLOAK_ADMIN","admin")
+            .WithEnvironment("KEYCLOAK_ADMIN_PASSWORD","admin")
+            .WithArgs("start-dev");
```

Comment:
`start-dev` is a good start, but that's purely for development environments, and is not suitable for production. 

See https://www.keycloak.org/server/configuration for more info.

Of course, using any other start command requires a database to be setup and for the respective connection string info to be passed along, which complicates this in a major way. I assume you're aware of that but wanted to add it here in case others were not.

Reply:
I agree with you, we should simply execute `start` command to simulate production. We can either add arguments to specify the desired command. Or, validate the environment through the builder and launch `start-dev` in the case of a development environment.

As for the database configuration, we could either add a reference or inject the connection string.

Reply:
Love it!

One thing that would be particularly nice is to reference an existing database connection, and pass that reference along to this extension here. I think that'd be really elegant from a DX perspective.

Reply:
I provide a progress report on the work I am carrying out.

In a development environment, the command `start-dev` is executed, in other cases, it is the `start` command.

I have added two additional optional arguments, the type of database, as well as the associated connection string. In the case where the database is not specified, it is identified through the connection string. For now, I have conducted tests only with PostgreSQL.

However, when I pass the reference of a database managed by Aspire using the `WithReference` method, I am unable to access the database. I retrieve the database connection string, but access is denied. I am trying to understand the reasons. I suppose it comes from the fact that the KeyCloak container depends on another container.